src/python/m5/ext/pystats/jsonloader.py

```diff
@@ -1,55 +1,72 @@
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
 from .simstat import SimStat
 from .statistic import Scalar, Distribution, Accumulator
 from .group import Group, Vector
 import json
 
-def json_loader(path: str) -> SimStat:
-    
+class JSONLoader(json.JSONDecoder):
+    """
+    Subclass of JSONDecoder that overrides 'object_hook'. Converts JSON object
+    into a SimStat object.
+
+    Usage
+    -----
+    ```
+    from m5.ext.pystats.jsonloader import JSONLoader
+
     with open(path) as f:
-        simstat_obj = json.load(f, object_hook=__json_to_simstat)
-
-    return simstat_obj
-
-def __json_to_simstat(d: dict) -> SimStat:
-    print(d)
-    if 'type' in d:
-        if d['type'] == 'Scalar':
-            return Scalar(
-                          value=d.get('value'),
-                          unit=d.get('unit'),
-                          description=d.get('description'),
-                          datatype=d.get('datatype')
-                         )
-        elif d['type'] == 'Distribution':
-            return Distribution(
-                                value=d.get('value'),
-                                min=d.get('min'),
-                                max=d.get('max'),
-                                num_bins=d.get('numBins'),
-                                bin_size=d.get('binSize'),
-                                sum=d.get('sum'),
-                                sum_squared=d.get('sumSquared'),
-                                underflow=d.get('underflow'),
-                                overflow=d.get('overflow'),
-                                logs=d.get('logs'),
-                                unit=d.get('unit'),
-                                description=d.get('description'),
-                                datatype=d.get('datatype')
-                               )
-        elif d['type'] == 'Accumulator':
-            return Accumulator(
-                                value=d.get('value'),
-                                count=d.get('count'),
-                                min=d.get('min'),
-                                max=d.get('max'),
-                                sumSquared=d.get('sum_squared'),
-                                unit=d.get('unit'),
-                                description=d.get('description'),
-                                datatype=d.get('datatype')
-                               )
-        elif d['type'] == 'Group':
-            return Group(**d)
-        elif d['type'] == 'Vector':
-            return Group(**d)
-    else:
-        return SimStat(**d)
\ No newline at end of file
+    simstat_object = json.load(f, cls=JSONLoader)
+    ```
+    """
```

Comment:
@powerjg What do you think of this interface, where caller must specify 'cls' argument?

Reply:
Sorry, not sure why this comment is so weirdly formatted. I'm referring to the 'Usage'